src/app/Shared/Services/Api.service.tsx

```diff
@@ -257,10 +257,10 @@ export class ApiService {
     );
   }
 
-  updateRule(rule: Rule): Observable<boolean> {
+  updateRule(rule: Rule, clean = true): Observable<boolean> {
     const headers = new Headers();
     headers.set('Content-Type', 'application/json');
-    return this.sendRequest('v2', `rules/${rule.name}`, {
+    return this.sendRequest('v2', `rules/${rule.name}?clean=${clean}`, {
```

Comment:
I was just looking at this actually. I think we'll be making much more use of query params in the future for properly doing searching/filtering, sorting, pagination, etc. on the backend, so may as well do a tiny bit of refactoring here to get ahead of it.

This could instead look like:

```typescript
return this.sendRequest('v2', `rules/${rule.name}?${new URLSearchParams({ clean: String(clean) /* needed because this requires values of string type */ })`, { ... });`

This way the query string isn't being manually created by hand and it's a little less error-prone. Maybe this should also be passed in to `sendRequest` as an optional argument instead of being part of the interpolated path string.
```

Reply:
Right that would be much cleaner! Added that!